src/compile/mark/text.ts

```diff
@@ -75,29 +107,29 @@ export namespace text {
     } else {
       applyColorAndOpacity(p, model);
     }
+  }
 
-
+  function text(fieldDefText: FieldDef, scaleName: string, config: Config): VgValueRef {
     // text
-    if (model.has(TEXT)) {
-      if (contains([QUANTITATIVE, TEMPORAL], model.fieldDef(TEXT).type)) {
-        const format = model.config().mark.format;
-        extend(p, formatMixins(model, TEXT, format));
-      } else {
-        p.text = { field: model.field(TEXT) };
+    if (fieldDefText) {
+      if (fieldDefText.field) {
+        if (contains([QUANTITATIVE, TEMPORAL], fieldDefText.type)) {
+          const format = config.mark.format;
+          extend(p, formatMixins(model, TEXT, format));
```

Comment:
This branch needs the model for the `formatMixins` function, how can we get around this?